### PR TITLE
Fix marshal v1.Taint to yaml panic

### PR DESCRIPTION
**Pull request by orangedeng**
**Problem:**
Marshal v1.Taint with timeAdded nil value to yaml panic.

**Solution:**
Use our own taint struct with omitempty flag of timeAdded field.

Related PR: 
https://github.com/rancher/types/pull/923

Issues:

https://github.com/rancher/rancher/issues/13972

### CODE REVIEW

**Comment by loganhz** [comment]
LGTM

File: pkg/controllers/management/node/controller.go [hunk 1]
```diff
@@ -122,7 +122,7 @@ func (m *Lifecycle) setupCustom(obj *v3.Node) {
 		},
 	}
 
-	obj.Spec.DesiredNodeTaints = obj.Status.NodeConfig.Taints
+	obj.Spec.DesiredNodeTaints = taints.GetTaintsFromStrings(obj.Spec.CustomConfig.Taints)
```
Comment (kinarashah) [inline pkg/controllers/management/node/controller.go]: isn't this supposed to be `obj.Status.NodeConfig`?

Comment (kinarashah) [reply pkg/controllers/management/node/controller.go]: Oh nvm, I saw panic happens only for v3

**Comment by alena1108** [comment]
@orangedeng can you confirm that we've tested the following case:

* While on rancher 2.2.x, user set the taints for nodes manually using kubectl
* Rancher upgraded to 2.3. User set taints should be preserved.
* Update cluster via UI. Make sure user set taints still remain.

**Comment by orangedeng** [comment]
Yes. Tested this case.
The logic for `nodetaints` controller is to add/remove taints to current set. The edit node taints API expect a desired node taints set for input. We will compare them with current taints on node and update v1.node.
We also have a controller to sync taints from v1.Node to v3.Node to keep the current set updated.

**Comment by alena1108** [comment]
LGTM

**Comment by orangedeng** [comment]
Rebased.

**Pull request merged**